packages/eas-cli/src/credentials/manager/SetUpAndroidBuildCredentials.ts

```diff
@@ -0,0 +1,60 @@
+import { Platform } from '@expo/eas-build-job';
+import assert from 'assert';
+
+import { AndroidActionType } from './Actions';
+import { CheckBuildProfileFlagAgainstEasJson } from './CheckBuildProfileFlagAgainstEasJson';
+import { Action } from './HelperActions';
+import { ManageAndroid } from './ManageAndroid';
+import { CredentialsContext, CredentialsContextProjectInfo } from '../context';
+
+export class SetUpAndroidBuildCredentials extends ManageAndroid {
+  constructor(
+    callingAction: Action,
+    projectDir: string,
+    private setUpBuildCredentialsWithProfileNameFromFlag: string
+  ) {
+    super(callingAction, projectDir);
+  }
+
+  override async runAsync(): Promise<void> {
+    const hasProjectContext = !!this.callingAction.projectInfo;
+    const buildProfile = hasProjectContext
+      ? await new CheckBuildProfileFlagAgainstEasJson(
+          this.projectDir,
+          Platform.ANDROID,
+          this.setUpBuildCredentialsWithProfileNameFromFlag
+        ).runAsync()
+      : null;
+    let projectInfo: CredentialsContextProjectInfo | null = null;
+    if (hasProjectContext) {
+      const { exp, projectId } = await this.callingAction.getDynamicPrivateProjectConfigAsync({
+        env: buildProfile?.env,
+      });
+      projectInfo = { exp, projectId };
+    }
+    const ctx = new CredentialsContext({
+      projectDir: process.cwd(),
+      projectInfo,
+      user: this.callingAction.actor,
+      graphqlClient: this.callingAction.graphqlClient,
+      analytics: this.callingAction.analytics,
+      env: buildProfile?.env,
+      nonInteractive: false,
+      vcsClient: this.callingAction.vcsClient,
+    });
+
+    let gradleContext;
+    if (ctx.hasProjectContext) {
+      assert(buildProfile, 'buildProfile must be defined in a project context');
+      gradleContext = await this.createProjectContextAsync(ctx, buildProfile);
+    }
+
+    if (this.setUpBuildCredentialsWithProfileNameFromFlag) {
+      await this.runProjectSpecificActionAsync(
+        ctx,
+        AndroidActionType.CreateKeystore, // Directly proceed to CreateKeystore
+        gradleContext
+      );
+    }
+  }
+}
```

Comment:
If someone has already set up their keystore, we don't want to make it again. 

instead of calling the `CreateKeystore` action, I'd consider calling the `SetUpBuildCredentials` action instead because it detects if credentials are already set up, then creates the keystore if not. It is the logic that runs on `eas build` so it will be great if we could reuse that https://github.com/expo/eas-cli/blob/94b217c16ed7fc540c439591ca9fd07b8ac1869e/packages/eas-cli/src/credentials/android/actions/SetUpBuildCredentials.ts#L31-L56